app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarController.java

```diff
@@ -186,10 +186,9 @@ public void deleteLectureFromBlacklist(int widgetId, String lecture) {
         return lectures;
     }
 
-    public CalendarItem getCalendarItemByStartAndEndTime(DateTime start, DateTime end) {
-        String startString = DateTimeUtils.INSTANCE.getDateTimeString(start);
-        String endString = DateTimeUtils.INSTANCE.getDateTimeString(end);
-        return calendarDao.getCalendarItemByStartAndEndTime(startString, endString);
+    @Nullable
+    public CalendarItem getCalendarItemById(String id) {
+        return calendarDao.getCalendarItemById(id);
```

Comment:
IMHO `getCalendarItemById()` shouldn't be nullable, but I guess you can't guarantee that that ID exists.
Also I think the id should probably be an int :man_shrugging: